### DISCUSSION TITLE
ascii to nc

### DISCUSSION THREAD

**Discussion by Deba57** [dtcenter/METplus#1613] [File I/O] [answered]
Hi,
I am using asciinc tool to read aeornet observation data using the following command in MET :  
ascii2nc \
/home/aod/tempr/20220401_20220430_SEDE_BOKER.lev15 \
/home/aod/tempr/output/sample2_ascii.nc \
-v 5

I see it is interpolating to 550nm, which is useful. But I am wondering under which output netcdf header one can find this interpolated data ? Also how to read that when validating against model forecast ? 
DEBUG 5: AeronetHandler::_readObservations() stationID: SEDE_BOKER from index 72  lat: 30.855  lon: 34.7822  elv: 480 from index 75
DEBUG 3: AeronetHandler::_readObservations() Confirmed AOD interpolation at 550: 0.712869 (0.712869)    440: 0.794593   675: 0.645283
DEBUG 3: AeronetHandler::_readObservations() Confirmed AOD interpolation at 550: 0.745546 (0.745546)    440: 0.83858    675: 0.669274

I have attached input data here (have to add .txt suffix for attachment in github).
[20220401_20220430_SEDE_BOKER.lev15.txt](https://github.com/dtcenter/METplus/files/8680995/20220401_20220430_SEDE_BOKER.lev15.txt)

Thanks
Deba

**Comment by j-opatz** [comment, marked as answer]
Hi Deba,

And thank you for your inquiries. I'm glad to see that the ascii2nc tool worked as expected; when looking for the interpolated values, they'll be stored in the resulting netCDF file under the `obs_val` variable.To provide a small snippet of the interpolated data, I ran your data through ascii2nc and found the following:

```
// global attributes:
                :MET_Obs_version = "1.02" ;
                :use_var_id = "true" ;
                :FileOrigins = "File / generated 20220513_173756 UTC on host seneca by the MET ascii2nc tool" ;
                :MET_version = "V10.1.1" ;
                :MET_tool = "ascii2nc" ;
data:

 obs_val = -999, 0.227149, 0.243412, -999, -999, 0.265431, -999, -999, -999, 
    -999, -999, -999, -999, -999, 0.29025, -999, -999, 0.299405, -999, -999, 
    0.303834, 0.295207, 0.583647, -999, -999, 81.26012, 6.325602, 11.2, 
    312.6264, 0.167541, -999, 1.0197, 0.8697, -999, -999, 0.6746, -999, -999, 
    -999, -999, -999, -999, -999, -999, 0.5005, -999, -999, 0.439479, -999, 
    -999, 0.3801, 0.340533, 0.936966, -999, -999, 0.2811797, -999, 0.229996, 
    0.246695, -999, -999, 0.269127, -999, -999, -999, -999, -999, -999, -999, 
    -999, 0.294126, -999, -999, 0.303331, -999, -999, 0.309008, 0.299516, 
    0.590897, -999, -999, 80.62933, 5.93277, 11.6, 312.6263, 0.167542, -999, 
    1.0197, 0.8697, -999, -999, 0.6746, -999, -999, -999, -999, -999, -999, 
    -999, -999, 0.5005, -999, -999, 0.439479, -999, -999, 0.3801, 0.340533, 
    0.936966, -999, -999, 0.2849857, -999, 0.233779, 0.250572, -999, -999,... 
```
Note that this is all of the variable types (e.g.  "AOD", "Precipitable_Water", "Solar_Zenith_Angle", "Optical_Air_Mass", "Sensor_Temperature", "Ozone", "NO2", "Exact_Wavelengths_of_AOD", "Exact_Wavelengths_of_PW") together, and would require further sorting to extract only 1 variable.

For your second question, 

> Also how to read that when validating against model forecast ?

I'll use an example with the point-stat tool, which assumes a gridded forecast field and point observation data. Take a look at the [Users Guide entry for point-stat](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#point-stat-tool): it's a great resource for where to start and troubleshooting for tools. 
You'll need to fill in a point-stat configuration file that fits your forecast and observation input file types/formats, plus any climatology data you'll use. Focusing on the observation dictionary, which is where your `sample2_ascii.nc` file will be read in, you'd use something like this to read in the `Precipitable_Water` variable 

```
obs = {
   field = [
      {
        name       = "Precipitable_Water";
        level      = [ "Z0" ];
 }
];
}
```
And you'd make a command line call with something like 

`point_stat fcst_file.fmt sample2_ascii.nc PointStatConfig_Sample`

Be sure to read in the Users Guide what's necessary for each type of file input (netCDF, Grib2, etc.) as that will determine how to fill in the fcst dictionary, as well as the climatology dictionaries.

Let us know if you have any further questions on this topic, or run into any roadblocks on this subject and we'll be happy to point you in the right direction.

**Reply by Deba57** [reply]
Thanks John. So after interpolation to 550nm for AOD, one can read that interpolated data following way in the config file (to validate against model forecast) : 
obs = {
      field = [
      {
        name       = "AOD";
        level      = [ "L550" ];
        GRIB1_ptv  = 141;
      }
   ];
 }

Please confirm.

**Reply by j-opatz** [reply]
That's correct; that should be what you need to make MET pull out AOD from the netCDF file.

But if there are any issues please let us know and we can get it straightened out.

**Reply by Deba57** [reply]
Great, thanks @j-opatz
